File: backend/app/schemas/workspace_validation.py
from pydantic import BaseModel


class ValidationWarning(BaseModel):
    type: str
    message: str
    detail: str | None = None


class ValidationError(BaseModel):
    type: str
    message: str
    detail: str | None = None


class ProjectStructureInfo(BaseModel):
    has_frontend: bool
    has_backend: bool
    has_source_folders: bool
    has_config_folder: bool
    has_assets_folder: bool
    has_documentation_folder: bool
    has_hidden_folders: bool
    notes: list[str]


class ConfigFilesSummary(BaseModel):
    has_package_json: bool
    has_requirements_txt: bool
    has_pyproject_toml: bool
    has_dockerfile: bool
    has_docker_compose: bool
    has_readme: bool
    has_env_example: bool
    files_found: list[str]
    files_missing: list[str]


class WorkspaceValidationResponse(BaseModel):
    workspace_status: str
    validation_result: str
    warnings: list[ValidationWarning]
    errors: list[ValidationError]
    project_structure: ProjectStructureInfo
    config_files_summary: ConfigFilesSummary
    ready_for_analysis: bool
    summary: str

File: frontend/src/types/workspace-validation.ts
export interface ValidationWarning {
  type: string;
  message: string;
  detail: string | null;
}

export interface ValidationError {
  type: string;
  message: string;
  detail: string | null;
}

export interface ProjectStructureInfo {
  has_frontend: boolean;
  has_backend: boolean;
  has_source_folders: boolean;
  has_config_folder: boolean;
  has_assets_folder: boolean;
  has_documentation_folder: boolean;
  has_hidden_folders: boolean;
  notes: string[];
}

export interface ConfigFilesSummary {
  has_package_json: boolean;
  has_requirements_txt: boolean;
  has_pyproject_toml: boolean;
  has_dockerfile: boolean;
  has_docker_compose: boolean;
  has_readme: boolean;
  has_env_example: boolean;
  files_found: string[];
  files_missing: string[];
}

export interface WorkspaceValidation {
  workspace_status: string;
  validation_result: string;
  warnings: ValidationWarning[];
  errors: ValidationError[];
  project_structure: ProjectStructureInfo;
  config_files_summary: ConfigFilesSummary;
  ready_for_analysis: boolean;
  summary: string;
}
